### PR TITLE
Add first steps to allow referees to provide feedback in the app

### PR DESCRIPTION
**Pull request by alexdesi**
## Context

This PR allows a referee to leave a reference to the candidate (1st part)
- send a new email with a tokenized URL
- check if the referee is allowed to see the page

## Link to Trello card

https://trello.com/c/B4tokVQU/§

### CODE REVIEW

File: app/views/referee_mailer/reference_request_by_google_form_email.text.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Dear <%= @reference.name %>,
+
+# Give a reference for <%= @candidate_name %>
+
+<%= @candidate_name %> put us in touch with you to get a reference for their teacher training application. They applied to:
+
+<% @application_form.application_choices.each do |application_choice| %>
+* <%= application_choice.provider.name %> - <%= application_choice.course.name %>
```
Comment (benilovj) [inline app/views/referee_mailer/reference_request_by_google_form_email.text.erb]: Do we want to have a method on `ApplicationChoice` that returns a human-readable ASCII description of the choice? That way this description can be evolved in one place... /cc @fofr

File: app/controllers/referee_interface/reference_controller.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      RequestLocals.store[:identity] = { reference_id: reference.id }
+      Raven.user_context(reference_id: reference.id)
+    end
+
+    def reference
+      @reference ||= ApplicationReference.find_by_unhashed_token(params[:token])
+    end
+
+    def check_referee_has_valid_token
+      render_404 unless reference
```
Comment (fofr) [inline app/controllers/referee_interface/reference_controller.rb]: 404 or 401?

File: app/views/referee_mailer/reference_request_by_google_form_email.text.erb (hunk 1 above)
Comment (tijmenb) [reply app/views/referee_mailer/reference_request_by_google_form_email.text.erb]: Let's pick that up in a separate PR, the choice description is not consistent across the emails anyway.

File: app/controllers/referee_interface/reference_controller.rb (hunk 2 above)
Comment (tijmenb) [reply app/controllers/referee_interface/reference_controller.rb]: 404

**Pull request merged**